core/openai.go

```diff
@@ -0,0 +1,110 @@
+package core
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+)
+
+var	inferURL = os.Getenv("INFER_URL")
+
+// Client represents a client for communicating with the OpenAI API.
+type OpenAIClient struct {
+	Token     string
+	Organization string
+	ModelID  string
+	httpClient *http.Client
+}
+
+// NewClient returns a new instance of a Client.
+func NewClient(organization, modelID string) *OpenAIClient {
+	token := os.Getenv("OPENAI_API_KEY")
+	if token == "" {
+		log.Fatalf("OPENAI_API_KEY environment variable is not set")
+	}
+	return &OpenAIClient{
+		Token: token,
+		Organization: organization,
+		ModelID: modelID,
+		httpClient: http.DefaultClient,
+	}
+}
```

Comment:
**(issue, blocking)** The function controls what shall not be controlled within its scope. 

```go
type OpenAPIConfig struct {
    token             string
    baseURL       string
    // it can be customised
    HttpClient      http.Client
}

func NewClient(token string) (*OpenAIClient, error) {
    if token == "" {
        return nil, errors.New("token must be provided")
    }
    return &OpenAIClient{
 		token: token,
		baseURL: inferURL,
		HttpClient: http.DefaultClient,
    }, nil
}
```

**(nitpick)** Do not put uncontrolled process termination within the function; early stop shall be implemented on the function's user discretion based on the returned error.